test/jit/test_backend_nnapi.py

```diff
@@ -29,73 +29,49 @@
 
 """
 Unit Tests for Nnapi backend with delegate
+Inherits most tests from TestNNAPI, which loads Android NNAPI models
+without the delegate API.
 """
 # This is needed for IS_WINDOWS or IS_MACOS to skip the tests.
 @unittest.skipIf(TEST_WITH_ROCM or IS_SANDCASTLE or IS_WINDOWS or IS_MACOS or IS_FBCODE,
                  "Non-portable load_library call used in test")
-class NnapiBackendPReLUTest(JitTestCase):
-    """
-    Test lowering a simple PRelU module to Nnapi backend.
-    TODO: After Nnapi delegate is finished T91991928,
-    add tests for running the module (currently only lowers the model)
-    """
+class TestNnapiBackend(TestNNAPI):
     def setUp(self):
         super().setUp()
-        # Load nnapi delegate library
-        torch_root = Path(__file__).resolve().parent.parent.parent
-        p = torch_root / 'build' / 'lib' / 'libnnapi_backend.so'
-        torch.ops.load_library(str(p))
-
-        # Nnapi set up
-        # Avoid saturation in fbgemm
-        torch.backends.quantized.engine = 'qnnpack'
 
-        libneuralnetworks_path = os.environ.get("LIBNEURALNETWORKS_PATH")
-        if libneuralnetworks_path:
-            ctypes.cdll.LoadLibrary(libneuralnetworks_path)
-            print("Will attempt to run NNAPI models.")
-            self.can_run_nnapi = True
-        else:
-            self.can_run_nnapi = False
-
-    def test_execution(self):
         # Save default dtype
         module = torch.nn.PReLU()
-        default_dtype = module.weight.dtype
+        self.default_dtype = module.weight.dtype
         # Change dtype to float32 (since a different unit test changed dtype to float64,
         # which is not supported by the Android NNAPI delegate)
         # Float32 should typically be the default in other files.
         torch.set_default_dtype(torch.float32)
-        module = torch.nn.PReLU()
-        args = torch.tensor([[1.0, -1.0, 2.0, -2.0]]).unsqueeze(-1).unsqueeze(-1)
 
-        # Trace and lower PreLu module
-        traced = torch.jit.trace(module, args)
-        compile_spec = {"forward": {"inputs": args}}
-        nnapi_model = torch._C._jit_to_backend("nnapi", traced, compile_spec)
+        # Load nnapi delegate library
+        torch_root = Path(__file__).resolve().parent.parent.parent
+        p = torch_root / 'build' / 'lib' / 'libnnapi_backend.so'
+        torch.ops.load_library(str(p))
```

Comment:
Can we just build this into the normal PyTorch build so we don't need a separate load step?  This doesn't seem portable.

Reply:
I don't quite know how to build this into the normal PyTorch build or what it means. I copied this step from what's in test_backends.py. @iseeyuan @raziel Any thoughts?

Reply:
I thought building an .so file and loading it from python is a format used for the backends, to show the flexibility of loading different backends as needed. But you are right, it's not portable for testing purpose. @Amyh11325 could you add an action item in your doc, so we could follow up when you finish the milestones?

Reply:
So, it's the plan to keep testing with the .so for the base infra tests, and then the specific backends tests to do it in the PyTorch build?